### PR TITLE
Detect & handle JaRule timeouts.

### PR DESCRIPTION
**Pull request by nomis52**
Also add the Return Code enums

### CODE REVIEW

File: tools/ja-rule/ja-rule.cpp [hunk 1]
```diff
@@ -292,80 +296,69 @@ class Controller {
...
-      OLA_INFO << "RC: " << static_cast<int>(return_code);
-      return;
-    }
-
     string response;
     if (!payload.empty()) {
        response.append(reinterpret_cast<const char*>(payload.data()),
                        payload.size());
     }
-    OLA_INFO << "Echo Reply (" << static_cast<int>(return_code) << "): "
```
Comment (peternewman) [inline tools/ja-rule/ja-rule.cpp]: Is it not worth leaving the Echo Reply bit, for what type of command it was?

File: tools/ja-rule/ja-rule.cpp [hunk 2]
```diff
@@ -292,80 +296,69 @@ class Controller {
...
   void AckCommandComplete(
       ola::usb::USBCommandResult result,
-      uint8_t return_code, uint8_t status_flags,
+      JaRuleReturnCode return_code, uint8_t status_flags,
       const ByteString &payload) {
     if (!CheckResult(result, status_flags)) {
       return;
     }
 
-    OLA_INFO << "ACK (" << static_cast<int>(return_code)
```
Comment (peternewman) [inline tools/ja-rule/ja-rule.cpp]: Again

File: tools/ja-rule/ja-rule.cpp [hunk 3]
```diff
@@ -292,80 +296,69 @@ class Controller {
...
-             << "): payload_size: " << payload.size();
-    OLA_INFO << "Flags (" << static_cast<int>(return_code) << "):";
```
Comment (peternewman) [inline tools/ja-rule/ja-rule.cpp]: Again

**Comment by peternewman** [comment]
Generally LGTM

File: tools/ja-rule/ja-rule.cpp (hunk 1 above)
Comment (nomis52) [inline tools/ja-rule/ja-rule.cpp]: If the command isn't an echo the result will be COMMAND_RESULT_CLASS_MISMATCH

File: tools/ja-rule/ja-rule.cpp (hunk 2 above)
Comment (nomis52) [inline tools/ja-rule/ja-rule.cpp]: Same

File: tools/ja-rule/ja-rule.cpp (hunk 3 above)
Comment (nomis52) [inline tools/ja-rule/ja-rule.cpp]: Same

File: tools/ja-rule/ja-rule.cpp (hunk 1 above)
Comment (peternewman) [inline tools/ja-rule/ja-rule.cpp]: I'm meaning more so that the output is actual useful/friendly for a human, but you'll see more in my upcoming PR.

**Pull request merged**